docs/docs/features/studio.md

```diff
@@ -0,0 +1,108 @@
+---
+title: ZMK Studio
+---
+
+:::warning[Alpha Feature]
+
+ZMK Studio support is in alpha. Although best efforts are being made, keeping compatibility during active development is not guaranteed.
+
+:::
+
+ZMK Studio provides runtime update functionality to ZMK powered devices, allowing users to change their keymap layers without flashing new firmware to their keyboards. Studio is still under active development, and is not yet ready for casual end user use.
+
+## Building
+
+Building for ZMK Studio involves two main additional items.
+
+- Build with the `studio-rpc-usb-uart` snippet to enable the endpoint used for ZMK Studio communication over USB.
+- Enable the `ZMK_STUDIO` Kconfig setting.
+
+### GitHub Actions
+
+First add a `studio-rpc-usb-uart` to the `snippet` property of your build configuration. For a split keyboard, you should do this _only_ for your central/left side, e.g.:
+
+```
+---
+include:
+  - board: nice_nano_v2
+    shield: corne_left
+    snippet: studio-rpc-usb-uart
+  - board: nice_nano_v2
+    shield: corne_right
+```
+
+Next, enable the `ZMK_STUDIO` Kconfig symbol, for example by adding the following line to your .conf file:
+
+```
+CONFIG_ZMK_STUDIO=y
+```
+
+### Local Build
+
+When building locally, use the `-S` parameter to include the `studio-rpc-usb-uart` snippet. Instead of adding it to your config file, you can also append the `ZMK_STUDIO` Kconfig as an additional CMake argument, e.g.:
+
+```bash
+west build -d build/cl_studio -b nice_nano_v2 \
+  -S studio-rpc-usb-uart -- -DSHIELD=corne_left -DCONFIG_ZMK_STUDIO=y
+```
+
+## Including Unreferenced Behaviors
+
+Normally, ZMK will only build and include behaviors that are referenced by your keymap and unused behavior will be skipped. However, ZMK Studio builds using the `studio-rpc-usb-uart` snippet will automatically define the `ZMK_BEHAVIORS_KEEP_ALL` value, which changes the approach and builds all possible behaviors into the firmware. This lets those behaviors be used in ZMK Studio.
```

Comment:
Note that there would be a name collision if someone were to create a behavior named "all". Not sure if that's likely enough to be worth changing this, but we could maybe add a word like "studio" somewhere before "keep" so it's in a different namespace?

Reply:
So... I'd originally used `STUDIO` in the naming, but in reality this isn't *really* a Studio specific thing, but a general way to keep behaviors in the DTS to ensure they're built, so I'm a bit resistant to re-introducing it.  I'm not sure it's *that* huge of a deal to have "ALL" be reserved, especially since the naming isn't *strictly* required to match 1-to-1.